### PR TITLE
[SpringBoot 2.3] Update hazelcast cache configuration, remove obsolete props

### LINKED ISSUE #11244: Update Hazelcast management-center to 4.0.x
##### **Overview of the feature request**

See:
- https://docs.hazelcast.org/docs/management-center/4.0/manual/html/index.html
- https://docs.hazelcast.org/docs/management-center/rn/index.html#4-0
- https://hazelcast.com/blog/hazelcast-imdg-4-0-is-released
- https://docs.hazelcast.org/docs/latest/manual/html-single/#migration-guides
- https://github.com/hazelcast/hazelcast/releases/tag/v4.0
- https://github.com/hazelcast/hazelcast/wiki/4.0
- https://github.com/hazelcast/hazelcast-hibernate5/releases/tag/v2.0.0

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Update hazelcast cache configuration, remove obsolete props, update client resources

Closes #11244 

----
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
close/re-open to trigger builds

File: generators/generator-constants.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ const DOCKER_COUCHBASE = 'couchbase:6.5.1';
 const DOCKER_CASSANDRA = 'cassandra:3.11.5'; // issues upgrading to 3.11.6
 const DOCKER_MSSQL = 'mcr.microsoft.com/mssql/server:2017-latest-ubuntu';
 const DOCKER_NEO4J = 'neo4j:4.0.4';
-const DOCKER_HAZELCAST_MANAGEMENT_CENTER = 'hazelcast/management-center:3.12.9'; // waiting for https://github.com/jhipster/generator-jhipster/issues/11244
+const DOCKER_HAZELCAST_MANAGEMENT_CENTER = 'hazelcast/management-center:4.0.2';
```
Comment (DanielFran) [inline generators/generator-constants.js:64]: Can you update to 4.0.3?

Comment (vishal423) [reply generators/generator-constants.js]: should be good now

**Comment by vishal423** [comment]
I think we are good on this PR. @murdos , do you see an issue If I merge it on the spring boot v2.3 branch?

**Comment by murdos** [comment]
If you're confident, no problem, feel free to merge it.

**Comment by vishal423** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/24757

**Comment by pascalgrimaud** [comment]
@vishal423 : approved

**Pull request merged**